### DISCUSSION TITLE
Please supply the dockerfiles/containerfiles used to produce the .NET images.

### DISCUSSION THREAD

**Discussion by abotelho-cbn** [dotnet/dotnet-docker#4031] [General]
### Please supply the dockerfiles/containerfiles used to produce the .NET images.

Please provide the dockerfiles/containerfiles which are used to add .NET to the Debian base images so that we may recreated the Microsoft-supplied .NET images from scratch.

Cheers,
Alex

**Comment by mthalman** [comment]
It depends on which on the following things to determine which Dockerfile:
* Image type (runtime, aspnet, sdk)
* .NET version
* OS version
* architecture

For example, you can find the ASP.NET Core Dockerfiles for .NET 6 on AMD64 at https://github.com/dotnet/dotnet-docker/blob/dfb7841b9f13d28c9c6c967198ac6085a98750f9/src/aspnet/6.0/bullseye-slim/amd64/Dockerfile.

The easiest way to find the associated Dockerfile is to go to the [.NET Docker Hub page](https://hub.docker.com/_/microsoft-dotnet), find the repository you want, and go to the Tag Listing table. That table will contain a link to the Dockerfile for each of the images, listed by the tag.

For example, here's a tag listing at https://hub.docker.com/_/microsoft-dotnet-aspnet/:

![image](https://user-images.githubusercontent.com/15789599/186220241-a742b8c7-a629-4a47-8ef0-4487fce467b9.png)

**Reply by abotelho-cbn** [reply]
Alright, so this may explain things:
![image](https://user-images.githubusercontent.com/81174902/186225324-4cc8c0d2-ab9c-4dcb-b679-face4f3b68f8.png)
Checking the tags was the first thing I did, but I cannot see Tags on the Microsoft images.

I just confirmed Tags are working in general by checking the official Ubuntu image, which displays Tags.

**Reply by mthalman** [reply]
You need to click the link for which repository you want to get the Dockerfile of. That will contain the tag listing.

**Reply by abotelho-cbn** [reply]
So I am seeing the tag listing in the description of the more specific repos, but still no Tags tab

**Reply by mthalman** [reply]
There is no tags "tab". The tag listing contains the links to the associated Dockerfiles. You can see that in the screenshot I posted earlier.

**Reply by abotelho-cbn** [reply]
Ah, that explains it.
